### DISCUSSION TITLE
[GSoC] 7. Add Video Anomaly Detection Inference to Gradio

### DISCUSSION THREAD

**Discussion by samet-akcay** [openvinotoolkit/openvino#16265] [Google Summer of Code]
# Short description
Currently, Anomalib has a Gradio inference interface. However only image inference is supported. This project would add video support to Anomalib's Gradio inference. The following is an overview of the project at a high level:

Enable OpenVINO's asynchronous inference mode in anomalib to accelerate video prediction inference.
Modify Anomalib's Gradio inference interface to support OpenVINO's async inference mode for video prediction.

> **OPTIONAL** 
Add a new algorithm for detecting video anomalies to Anomalib. Note that a portion of the workload may overlap with another project, as both require implementing the async inference capabilities of OpenVINO. This would be a great opportunity to learn how to collaborate with other developers and resolve git merge conflicts, which are common in everyday software development.

# Expected outcomes
A new Gradio inference UI that accepts both image and video inputs and performs async OpenVINO inference based on a video anomaly detection model that is optionally added by the applicant.

# Skills required/preferred
DL (Elementary), Python Refactoring (Intermediate), Gradio (Elementary), PyTorch Lightning (Elementary)

# Mentors
Samet Akcay, Dick Ameln

# Size of project
175 or 350 hours (depending on working on the optional step)

# Difficulty
Medium/Hard (depending on choosing the optional step)

**Comment by sahilpmehra** [comment]
Hi @samet-akcay,

I would be very interested in contributing to this project idea for GSoC'23. I am almost done with my prerequisite task and would like to get started with any initial tasks/issues that I could solve related to this project. Could you help me with the same?

Thanks!

**Reply by samet-akcay** [reply]
Hi @sahilpmehra, thank you for your interest in this project! We are currently tagging some of the anomalib [issues](https://github.com/openvinotoolkit/anomalib/issues) as `Good first issue` so the applicants can work on. I'll provide an update here once it is done. Cheers!

**Reply by sahilpmehra** [reply]
Awesome! Thanks for the heads up @samet-akcay. Looking forward to it.
